AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -696,58 +834,148 @@ public Task<List<PositionStatusEventArgs>> RequestPositions(CancellationToken ca
+                this.clientSocket,
+                tokenSource.Token);
         }
 
         /// <summary>
-        /// Request security definition parameters.
-        /// This is mainly used for finding strikes, multipliers, exchanges, and expirations for options contracts.
+        /// Set the type for the market data feed
         /// </summary>
-        /// <param name="underlyingSymbol">The underlying symbol</param>
-        /// <param name="exchange">The exchange</param>
-        /// <param name="underlyingSecType">The underlying security type</param>
-        /// <param name="underlyingConId">The underlying contract ID, retrieved from the Contract Details call</param>
+        /// <param name="marketDataTypeId">The feed level</param>
+        public void RequestMarketDataType(int marketDataTypeId)
+        {
+            this.clientSocket.RequestMarketDataType(marketDataTypeId);
+        }
+
+        /// <summary>
+        /// Get the PnL of the account.
+        /// </summary>
+        /// <param name="accountCode">The account code</param>
+        /// <param name="modelCode">The model code</param>
+        /// <returns>The PnL account update event from TWS.</returns>
+        public Task<PnLEventArgs> RequestPnL(
+            string accountCode,
+            string modelCode)
+        {
+            // Set the operation to cancel after 5 seconds
+            CancellationTokenSource tokenSource = new CancellationTokenSource(5000);
+            return this.RequestPnL(accountCode, modelCode, tokenSource.Token);
+        }
+
+        /// <summary>
+        /// Get the PnL of the account.
+        /// </summary>
+        /// <param name="accountCode">The account code</param>
+        /// <param name="modelCode">The model code</param>
         /// <param name="cancellationToken">The cancellation token used to cancel the request</param>
-        /// <returns>The security definitions for options</returns>
-        public Task<List<SecurityDefinitionOptionParameterEventArgs>> RequestSecurityDefinitionOptionParameters(
-            string underlyingSymbol,
-            string exchange,
-            string underlyingSecType,
-            int underlyingConId,
+        /// <returns>The PnL account update event from TWS.</returns>
+        public Task<PnLEventArgs> RequestPnL(
+            string accountCode,
+            string modelCode,
             CancellationToken cancellationToken)
         {
-            var taskSource = new TaskCompletionSource<List<SecurityDefinitionOptionParameterEventArgs>>();
+            PnLEventArgs pnlUpdateEvent = null;
+            var taskSource = new TaskCompletionSource<PnLEventArgs>();
+            EventHandler<PnLEventArgs> pnlUpdateEventHandler = null;
 
-            List<SecurityDefinitionOptionParameterEventArgs> securityDefinitionEvents = new List<SecurityDefinitionOptionParameterEventArgs>();
+            pnlUpdateEventHandler = (sender, args) =>
+            {
+                pnlUpdateEvent = args;
```

Comment:
https://interactivebrokers.github.io/tws-api/pnl.html

Looks like this is a subscription method, so you don't want to return the PNL event immediately when it's fired.

Reply:
Ok.